docs/examples/howto/add_new_algorithm.rst

```diff
@@ -0,0 +1,174 @@
+Add a new algorithm
+===================
+
+Purpose
+-------------
+
+The main part of ACTS code is located in `Core` packages. 
+For them to be used in standalone ACTS an algorithm is needed.
+Also, new ideas need to be first developed as algorithms.
+Typically the essential parts of the code are then moved to the `Core` packages 
+and the algorithm only executes it.
+
+
+
+Code Organisation
+------------------
+
+Algorithms reside in `Examples/Algorithms` are of the repository.
+The code is split into header with the algorithm class declaration 
+and a source file containing the implementation.
+
+Assuming that you want to experiment with a new seeding algorithm the files to add would be:
+`Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/MySeedingAlgorithm.h`
+and 
+`Examples/Algorithms/TrackFinding/src/MySeedingAlgorithm.h`
+
+Algorithm Class
+---------------
+The algorithm class has to inherit from {class}`ActsExamples::BareAlgorithm`
+and thus implement single method:
+
+.. code-block:: cpp
+
+    ProcessCode execute(const AlgorithmContext& ctx) const final override;
+
+.. hint:: There are other types of algorithms for reading inputs from files
+    or for saving data to files. All these interfaces can be found in 
+    `Examples/Framework/include/ActsExamples/Framework`.
+
+.. important:: The constructor should ideally follow certain rules. See section on configuration below.
+
+The `execute` method will be called once for each event. 
```

Comment:
Suggest: The execute method will be called once for each event. Other methods can also be added to your class. It is good to remember that the algorithmic code in ACTS is typically stateless, and if the state needs to be passed between the calls it should be done explicitly. For instance if the event processing is best organized with such methods:

Note - does this really explain what is meant by stateless and passing states explicitly? I worry that this may not be super clear to casual readers? I guess it depends on our audience